### PR TITLE
AMLS-5094 | Trading Premises

### PR DESCRIPTION
**Pull request by owen-opencast**
Updated Trading Premises auth and some related services.

## How Has This Been Tested?

Passes spec tests, acceptance tests, manual testing and coverage.

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [ ] Requires acceptance test run.
- [x] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: test/controllers/tradingpremises/TradingPremisesAddControllerSpec.scala [hunk 1]
```diff
@@ -46,18 +45,18 @@ class TradingPremisesAddControllerSpec extends AmlsSpec with PropertyChecks with
     "load What You Need successfully when displayGuidance is true" in new Fixture {
 
       val BusinessActivitiesModel = BusinessActivities(Set(MoneyServiceBusiness, TrustAndCompanyServices, TelephonePaymentService))
-      val mockCacheMap = mock[CacheMap]
+      override val mockCacheMap = mock[CacheMap]
```
[the remark this answers is not in the document]
Comment (owen-opencast) [reply test/controllers/tradingpremises/TradingPremisesAddControllerSpec.scala]: The overrides were necessary, I was getting compilation errors without them.

[the remark this answers is not in the document]
Comment (owen-opencast) [reply test/controllers/tradingpremises/TradingPremisesAddControllerSpec.scala]: Actually, I could remove these after I removed DependencyMocksNewAuth...

**Comment by owen-opencast** [comment]
I've made the suggested changes.

File: test/controllers/tradingpremises/TradingPremisesAddControllerSpec.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (slavvco) [reply test/controllers/tradingpremises/TradingPremisesAddControllerSpec.scala]: ok, whatever you decide it's good :)

**Comment by slavvco** [comment]
merging to auth-wip, integration testing later

**Pull request merged**